onnxruntime/core/mlas/lib/amd64/ErfKernelFma3.asm

```diff
@@ -0,0 +1,567 @@
+
+LComputeErf4x8Loop:
+        vbroadcastss ymm15,ErfConstants.ErfNegZero[rax]
+        vmovups      ymm0,YMMWORD PTR [rcx]        ; original input vx0
+        vmovups      ymm1,YMMWORD PTR 32[rcx]      ; original input vx1
+        vmovups      ymm2,YMMWORD PTR 64[rcx]      ; original input vx2
+        vmovups      ymm3,YMMWORD PTR 96[rcx]      ; original input vx3
+
+        vbroadcastss ymm14,ErfConstants.ErfMaxInt[rax]
+
+        vandps       ymm4,ymm0,ymm15               ; vsign0
+        vandps       ymm5,ymm1,ymm15               ; vsign1
+        vandps       ymm6,ymm2,ymm15               ; vsign2
+        vandps       ymm7,ymm3,ymm15               ; vsign3
+
+        vbroadcastss ymm8,ErfConstants.ErfSMALL_P0[rax]
+        vmovups      YMMWORD PTR ErfKernelFrame.ErfBuffer0[rsp],ymm4
+        vmovups      YMMWORD PTR 32+ErfKernelFrame.ErfBuffer0[rsp],ymm5
+        vmovups      YMMWORD PTR 64+ErfKernelFrame.ErfBuffer0[rsp],ymm6
+        vmovups      YMMWORD PTR 96+ErfKernelFrame.ErfBuffer0[rsp],ymm7
+
+        vmovaps      ymm9,ymm8
+        vmovaps      ymm10,ymm8
+        vmovaps      ymm11,ymm8
+
+        vandps       ymm0,ymm0,ymm14               ; abs(vx0)  va0
+        vandps       ymm1,ymm1,ymm14               ; abs(vx1)  va1
+        vandps       ymm2,ymm2,ymm14               ; abs(vx2)  va2
+        vandps       ymm3,ymm3,ymm14               ; abs(vx3)  va3
+
+        vbroadcastss ymm15,ErfConstants.ErfSMALL_P1[rax]
+        vmulps       ymm4,ymm0,ymm0                ; vs0 (square)
+        vmulps       ymm5,ymm1,ymm1                ; vs1
+        vmulps       ymm6,ymm2,ymm2                ; vs2
+        vmulps       ymm7,ymm3,ymm3                ; vs3
+
+        vbroadcastss ymm14,ErfConstants.ErfSMALL_P2[rax]
+        vfmadd213ps  ymm8,ymm4,ymm15
+        vfmadd213ps  ymm9,ymm5,ymm15
+        vfmadd213ps  ymm10,ymm6,ymm15
+        vfmadd213ps  ymm11,ymm7,ymm15
+
+        vbroadcastss ymm13,ErfConstants.ErfSMALL_P3[rax]
+        vfmadd213ps  ymm8,ymm4,ymm14
+        vfmadd213ps  ymm9,ymm5,ymm14
+        vfmadd213ps  ymm10,ymm6,ymm14
+        vfmadd213ps  ymm11,ymm7,ymm14
+
+        vbroadcastss ymm15,ErfConstants.ErfSMALL_P4[rax]
+        vfmadd213ps  ymm8,ymm4,ymm13
+        vfmadd213ps  ymm9,ymm5,ymm13
+        vfmadd213ps  ymm10,ymm6,ymm13
+        vfmadd213ps  ymm11,ymm7,ymm13
+
+
+        vbroadcastss ymm14,ErfConstants.ErfSMALL_P5_Minus_One[rax]
+        vfmadd213ps  ymm8,ymm4,ymm15
+        vfmadd213ps  ymm9,ymm5,ymm15
+        vfmadd213ps  ymm10,ymm6,ymm15
+        vfmadd213ps  ymm11,ymm7,ymm15
+
+        vfmadd213ps  ymm8,ymm4,ymm14
+        vfmadd213ps  ymm9,ymm5,ymm14
+        vfmadd213ps  ymm10,ymm6,ymm14
+        vfmadd213ps  ymm11,ymm7,ymm14
+
+        vbroadcastss ymm12,ErfConstants.ErfSplitBoundary[rax]
+        vfmadd213ps  ymm8,ymm0,ymm0
+        vfmadd213ps  ymm9,ymm1,ymm1
+        vfmadd213ps  ymm10,ymm2,ymm2
+        vfmadd213ps  ymm11,ymm3,ymm3
+
+        vcmpps       ymm4,ymm0,ymm12,30            ; vmask0
```

Comment:
vcmpps,30 is Intel specific. AMD only implements the first 8 predicates. Can you invert and use the vcmpltps instead?

Reply:
Cool to let me know this. change to vcmpleps.

Reply:
use vcmpgtps as cross all related code.